lighthouse-core/config/config.js

```diff
@@ -340,27 +339,50 @@ class Config {
     return config;
   }
 
- /**
-  * Return IDs of top-level aggregations from a config. Used by tools driving lighthouse-core
-  * @param {{aggregations: !Array<{name: string}>}} Lighthouse config object.
-  * @return {!Array<string>}  Name values of aggregations within
-  */
-  static getAggregations(config) {
-    return config.aggregations.map(agg => ({
-      name: agg.name,
-      id: agg.id
-    }));
+  /**
+   * @param {!Object<{audits: !Array<{id: string}>}>} categories
+   * @param {Array<string>=} categoryIds
+   * @param {Array<string>=} auditIds
+   * @return {!Object<{audits: !Array<{id: string}>}>}
+   */
+  static filterCategoriesAndAudits(categories, categoryIds = [], auditIds = []) {
+    categories = Object.assign({}, categories);
+
+    Object.keys(categories).forEach(categoryId => {
+      if (categoryIds.includes(categoryId)) {
+        return;
+      }
+
+      const category = Object.assign({}, categories[categoryId]);
+      category.audits = category.audits.filter(item => auditIds.includes(item.id));
+      if (category.audits.length) {
+        categories[categoryId] = category;
+      } else {
+        delete categories[categoryId];
```

Comment:
gotta make sure the path executed 5 times is super-opt right? ;) it's gone anyway by ditching audits